### ISSUE TITLE
Some games run for >500 turns under UT

### ISSUE THREAD

**Issue by arr28** [SanchoGGP/ggp-base#378]
Affects...

- selectivesukoshi
- sudoku

These work fine when running for real, but fail under UT.

```
junit.framework.AssertionFailedError: Game running for >500 turns!
at org.ggp.base.test.PuzzleBase.testPuzzle(PuzzleBase.java:228)
```

The UT (`PuzzleBase`) drives Sancho and, once it has submitted the move, calls `utWillBeTerminal`.

```java
  // Methods for use by UTs only
  public boolean utWillBeTerminal() throws TransitionDefinitionException
  {
    MachineState lState = getStateMachine().getNextState(getCurrentState(),
                                                         Collections.singletonList(mLastMove));
    return getStateMachine().isTerminal(lState);
  }
```

This used to work fine.  Not sure what's going wrong now.

**Comment by arr28** [comment]
That's odd - it works under UT on my PC.  Looks like it's only under snap-ci that it fails.  Related to number of threads??

**Comment by arr28** [comment]
Was fixed by the fix for #377.  Not worrying about it.
